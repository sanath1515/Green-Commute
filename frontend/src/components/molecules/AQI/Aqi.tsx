import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Typography } from "@material-ui/core";
import { AqiData, jobCountData } from "../../../Constants";
import Count from "../Jobcount/Count";
import theme from "../../../theme/theme";
import { useSelector } from "react-redux";
import { greenState } from "../../../store/reducers";

const useStyles = makeStyles({
  root: {
    width: "650px",
    height: "850px",
    backgroundColor: "#e7fce0",
    alignItems: "center",
    alignContent: "center",
    display: "flex",
    flexDirection: "column",
  },
  image: {
    flexGrow: 0,
    marginBottom: "20px",
    objectFit: "contain",
    position: "relative",
    top: "150px",
  },
  text: {
    //no number
    margin: "10px 0 0",
    color: "#19293b",
    position: "relative",
    top: "225px",
    width: 500,
    textAlign: "center",
    marginLeft: "auto",
    marginRight: "auto",
  },
  text1: {
    margin: "40px 0 0",
    color: "#19293b",
    position: "relative",
    top: "150px",
    width: 500,
    textAlign: "center",
    marginLeft: "auto",
    marginRight: "auto",
  },
  number: {
    position: "relative",
    top: "175px",
    marginLeft: "auto",
    marginRight: "auto",
    fontSize: "100px",
    color: "#19293b",
    width: "auto",
    height: "122px",
    fontWeight: "normal",
  },
  mock: {
    display: "flex",
    justifyContent: "space-around",
    position: "relative",
    top: "150px",
    width: "500px",
    marginTop: theme.spacing(5),
    fontWeight: 300,
  },
  job: {
    // marginLeft:theme.spacing()
  },
});

export type AqiProps = {
  step: number;
  number?: string;
};

const Aqi: React.FC<AqiProps> = ({ step, number }) => {
  const classes = useStyles();

  const workLocations: string[] = useSelector<
    greenState,
    greenState["worklocations"]
  >((state) => {
    return state.worklocations;
  });

  const aqis = ["830", "920"];

  const jobs = workLocations.slice(0, 2).map((temp, index) => (
    <div className={classes.job}>
      <Count city={temp} number={aqis[index % 2]} />
    </div>
  ));
  return (
    <div className={classes.root}>
      <img className={classes.image} src={AqiData[step - 1].img} />
      {number && step !== 2 && (
        <Typography className={classes.number} variant="h3">
          {number}
        </Typography>
      )}
      {number && step === 2 && <div className={classes.mock}>{jobs}</div>}
      <Typography
        className={!number ? classes.text : classes.text1}
        variant="h3"
      >
        {number ? AqiData[step - 1].textWithNum : AqiData[step - 1].textNoNum}
      </Typography>
    </div>
  );
};

export default Aqi;
