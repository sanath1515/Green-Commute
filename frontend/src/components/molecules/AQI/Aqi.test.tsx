/**
 * @jest-environment jsdom
 */

import { render, screen, cleanup } from "@testing-library/react";
import Aqi, { AqiProps } from "./Aqi";

afterEach(cleanup);

const data: AqiProps = {
  step: 1,
};

describe("Aqi Testing", () => {
  it("Checking The title", () => {
    render(<Aqi {...data} />);

    expect(
      screen.getByText("Enter location to know Time Air Quality Index (Aqi)")
    ).toBeInTheDocument();
  });

  it("Matches the snapshot", () => {
    const { container } = render(<Aqi {...data} />);
    expect(container).toMatchSnapshot();
  });
});
