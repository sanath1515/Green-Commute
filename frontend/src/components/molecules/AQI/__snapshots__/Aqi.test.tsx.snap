// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aqi Testing Matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-8"
  >
    <img
      class="makeStyles-image-9"
      src="https://cdn.zeplin.io/5fa28fcbff79c58958e4e1a6/assets/ae5e266d-f03e-4732-af5b-a8ebdac94494.svg"
    />
    <h3
      class="MuiTypography-root makeStyles-text-10 MuiTypography-h3"
    >
      Enter location to know Time Air Quality Index (Aqi)
    </h3>
  </div>
</div>
`;
